version: "3.8"

services:
  nginx:
    image: ${NGINX_IMAGE_NAME}:${NGINX_IMAGE_TAG}
    configs:
      - source: default_conf
        target: "/etc/nginx/conf.d/default.conf"
      - source: nginx_conf
        target: "/etc/nginx/nginx.conf"
    deploy:
      labels:
        - "traefik.enable=true"
        - "traefik.http.routers.nginx.entrypoints=web"
        - "traefik.http.routers.nginx.rule=${NGINX_URLS}"
        - "traefik.http.routers.nginx.service=nginx"
        - "traefik.http.services.nginx.loadbalancer.server.port=${NGINX_PORT}"
      placement:
        constraints:
          - "node.labels.nginx == true"
      replicas: ${REPLICAS}
      update_config:
        order: "start-first"
        parallelism: 1
    healthcheck:
      test:
        [
          "CMD-SHELL",
          "curl -fsSL localhost:${NGINX_PORT} || exit 1",
        ]
      interval: 30s
      retries: 3
      start_period: 30s
      timeout: 30s
    logging:
      driver: json-file
      options:
        max-file: "3"
        max-size: "10m"
    networks:
      - finding_aid_discovery
      - traefik

configs:
  default_conf:
    external: true
    name: "${NGINX_DEFAULT_CONF}"
  nginx_conf:
    external: true
    name: "${NGINX_CONF}"

networks:
  finding_aid_discovery:
    external: true
  traefik:
    external: true
